src/compute_horde_prompt_solver/prompt_solver.py

```diff
@@ -0,0 +1,208 @@
+        ready_to_terminate_event: mp.Event,
+        config: Config,
+):
+    start_server_event.wait()
+
+    app = Flask("compute_horde_prompt_solver")
+
+    @app.route("/health")
+    def server_healthcheck():
+        return {"status": "ok"}
+
+    @app.route("/execute-job", methods=["POST"])
+    def execute_job():
+        try:
+            from flask import request
+
+            seed_raw = request.json.get("seed")
+            seed = int(seed_raw)
+            seed_queue.put(seed)
+            result = result_queue.get(timeout=TIMEOUT)
+            return jsonify(result)
+        finally:
+            # The seed_queue.put(seed) can fail (request not having int seed etc.),
+            # so we always put a None to make sure process is terminated when the view returns.
+            seed_queue.put(None)
+
+    @app.route("/terminate")
+    def terminate():
+        ready_to_terminate_event.set()
+        return {"status": "ok"}
+
+    app.run(
+        host="0.0.0.0",
+        port=config.server_port,
+        debug=False,
+    )
+
+
+class BaseSolver(abc.ABC):
+    def __init__(
+            self,
+            provider: BaseLLMProvider,
+            config: Config
+    ):
+        self.provider = provider
+        self.config = config
+
+    def process_file(self, input_file, sampling_params):
+        with open(input_file, "r") as f:
+            prompts = [line.strip() for line in f if line.strip()]
+
+        responses = self.provider.generate_responses(prompts, sampling_params)
+
+        output_file = self.config.output_dir / f"{input_file.stem}.json"
+        self.save_output_file(responses, output_file)
+
+    def save_output_file(self, responses: Dict[str, str], output_file: pathlib.Path):
+        with open(output_file, "w") as f:
+            json.dump(responses, f, indent=2)
+
+    def get_sampling_params(self, seed):
+        set_deterministic(seed)
+
+        return SamplingParams(
+            max_tokens=self.config.max_tokens,
+            temperature=self.config.temperature,
+            top_p=self.config.top_p,
+            seed=seed,
+        )
+
+    @abc.abstractmethod
+    def run(self): ...
+
+
+class CLISolver(BaseSolver):
+
+    def run(self):
+        self.config.output_dir.mkdir(parents=True, exist_ok=True)
+
+        sampling_params = self.get_sampling_params(self.config.seed)
+
+        for input_file in self.config.input_files:
+            self.process_file(input_file, sampling_params)
+
+
+class HttpSolver(BaseSolver):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.start_server_event = mp.Event()
+        self.seed_queue = mp.Queue()
+        self.result_queue = mp.Queue()
+        self.ready_to_terminate_event = mp.Event()
+        self.response_hashes: Dict[str, str] = {}
+
+    def save_output_file(self, responses: Dict[str, str], output_file: pathlib.Path):
+        response_body = json.dumps(responses, indent=2).encode()
```

Comment:
I would use sort_keys=True - to make sure hash is always the same - I would not relay on responses dict order... unless we have contract saying it will return always the same order - then maybe mention it here

Reply:
ok - I see now that the order is controlled - still I like explicit hash computation of dict with sort_keys - your choice

Reply:
you're right, I didn't think about that cause I focused on our most important case of a single key in this dict